ports/stm32/boards/plli2svalues.py

```diff
@@ -166,35 +169,41 @@ def main():
             break
 
     if mcu_series in mcu_support_plli2s:
-        if len(argv) != 2:
+        if len(argv) not in (1, 2):
             print("usage: pllvalues.py [-c] [-m <mcu_series>] <hse in MHz> <pllm in MHz>")
             sys.exit(1)
 
         if argv[0].startswith("hse:"):
-            # extract HSE_VALUE from header file
+            hse = int(argv[0][len("hse:") :])
+
+        if argv[0].startswith("pllm:"):
+            pllm = int(argv[0][len("pllm:") :])
+
+        if argv[0].startswith("file:"):
+            # extract HSE_VALUE and from header file
             (hse,) = search_header(
-                argv[0][len("hse:") :],
+                argv[0][len("file:") :],
                 r'#include "(boards/[A-Za-z0-9_./]+)"',
-                r"#define +(HSE_VALUE) +\((\(uint32_t\))?([0-9]+)\)",
```

Comment:
I don't think this include regex and its associated logic in `search_header` is needed anymore.  That simplifies things, and the `vals` argument is then also not needed.

Reply:
Ah, yes I follow what you mean; the recursive search really isn't needed at all now that it's parsing the pre-processed / merged headers. Stripping that out certainly simplifies the logic!